pkg/controller/home/home_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package home_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/exposure-notifications-verification-server/internal/browser"
+	"github.com/google/exposure-notifications-verification-server/internal/envstest"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+
+	"github.com/chromedp/chromedp"
+)
+
+func TestHome(t *testing.T) {
+	t.Parallel()
+
+	harness := envstest.NewServer(t)
+
+	realm, err := harness.Database.FindRealm(1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	admin := &database.User{
+		Email:       "admin@example.com",
+		Name:        "Admin",
+		Realms:      []*database.Realm{realm},
+		AdminRealms: []*database.Realm{realm},
+	}
+	if err := harness.Database.SaveUser(admin, database.System); err != nil {
+		t.Fatal(err)
+	}
+
+	browserCtx := browser.NewContext(t, true)
+
+	taskCtx, done := context.WithTimeout(browserCtx, 30*time.Second)
+	defer done()
+
+	var code string
+	if err := chromedp.Run(taskCtx,
+		chromedp.Navigate(`http://`+harness.Server.Addr()),
+		Login("admin@example.com", "Password"),
+
+		// Post-login action is /home.
+		chromedp.WaitVisible(`body#home`, chromedp.ByQuery),
```

Comment:
What if there's a a "verify with text message" screen? Not sure how `WaitVisible` handles it.

Reply:
NVM, I see below.

Reply:
Yea, this is the first pass. I want to do some more thinking on how to "login" since it will be required for all journeys. I'm wondering if we can curate and inject a cookie instead of depending on firebase...